---
string base_path
string map_location